[Bug] Error: Evaluation failed: TypeError: pQuerySelectorAll is not a function

## Describe the bug

When running accessibility-insights-scan on portal.azure.com URLs with authentication, scanner fails with "Error: pQuerySelectorAll is not a function".

## Area

accessibility-insights-scan
accessibility-insights-crawler

## To Reproduce

Steps to reproduce the behavior:

Run ai-scan with authentication on the url https://portal.azure.com, e.g. `node .\packages\cli\dist\ai-scan-cli.js --url https://portal.azure.com --serviceAccountName $ACCOUNT_NAME --serviceAccountPassword $ACCOUNT_PASS --authType AAD --crawl`

## Expected behavior

The crawler successfully scans the URL.

## Log files

From a test pipeline using the Accessibility Insights for ADO extension:
```
2023-04-21T17:33:32.4813223Z ##[section]Starting: case 12: url, azure portal auth test
2023-04-21T17:33:32.4818816Z ==============================================================================
2023-04-21T17:33:32.4819227Z Task         : Accessibility Insights Azure DevOps Task
2023-04-21T17:33:32.4819477Z Description  : Scan for accessibility issues in an Azure DevOps pipeline
2023-04-21T17:33:32.4819800Z Version      : 3.0.9
2023-04-21T17:33:32.4820033Z Author       : Accessibility Insights
2023-04-21T17:33:32.4820316Z Help         : For help with configuring and troubleshooting this task, see https://aka.ms/accessibility-insights-faq#azure-devops-extension
2023-04-21T17:33:32.4820669Z ==============================================================================
2023-04-21T17:33:32.6828065Z ##[group]Installing runtime dependencies
2023-04-21T17:33:32.6836034Z ##[debug]Using node from /usr/local/vss-agent/3.218.0/externals/node16/bin/node
2023-04-21T17:33:32.6837943Z ##[debug]Using bundled yarn from /mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/.yarn/releases/yarn-3.4.1.cjs
2023-04-21T17:33:33.1307492Z ➤ YN0000: ┌ Resolution step
2023-04-21T17:33:33.2744215Z ➤ YN0002: │ @fluentui/font-icons-mdl2@npm:8.4.13 doesn't provide @types/react (p46e09), requested by @fluentui/utilities
2023-04-21T17:33:33.2746711Z ➤ YN0002: │ @fluentui/font-icons-mdl2@npm:8.4.13 doesn't provide react (pc3e60), requested by @fluentui/utilities
2023-04-21T17:33:33.2747865Z ➤ YN0002: │ @fluentui/style-utilities@npm:8.7.12 doesn't provide @types/react (p1c0a5), requested by @fluentui/utilities
2023-04-21T17:33:33.2749370Z ➤ YN0002: │ @fluentui/style-utilities@npm:8.7.12 doesn't provide @types/react (p1f230), requested by @fluentui/theme
2023-04-21T17:33:33.2750468Z ➤ YN0002: │ @fluentui/style-utilities@npm:8.7.12 doesn't provide react (pd254b), requested by @fluentui/utilities
2023-04-21T17:33:33.2751353Z ➤ YN0002: │ @fluentui/style-utilities@npm:8.7.12 doesn't provide react (p716df), requested by @fluentui/theme
2023-04-21T17:33:33.2752282Z ➤ YN0002: │ accessibility-insights-report@npm:4.6.3 doesn't provide @types/react (pcd13e), requested by @fluentui/react
2023-04-21T17:33:33.2755178Z ➤ YN0002: │ accessibility-insights-report@npm:4.6.3 doesn't provide @types/react-dom (p6fea8), requested by @fluentui/react
2023-04-21T17:33:33.2757392Z ➤ YN0002: │ accessibility-insights-report@npm:4.6.4 doesn't provide @types/react (p6edd6), requested by @fluentui/react
2023-04-21T17:33:33.2759035Z ➤ YN0002: │ accessibility-insights-report@npm:4.6.4 doesn't provide @types/react-dom (p17015), requested by @fluentui/react
2023-04-21T17:33:33.2760229Z ➤ YN0060: │ accessibility-insights-scan@npm:1.7.0 provides puppeteer (p62687) with version 19.8.5, which doesn't satisfy what @axe-core/puppeteer requests
2023-04-21T17:33:33.2762081Z ➤ YN0060: │ accessibility-insights-scan@npm:1.7.0 provides puppeteer (p07c1d) with version 19.8.5, which doesn't satisfy what apify requests
2023-04-21T17:33:33.2765175Z ➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
2023-04-21T17:33:33.2770684Z ➤ YN0000: └ Completed
2023-04-21T17:33:33.3180984Z ➤ YN0000: ┌ Fetch step
2023-04-21T17:33:33.6548686Z ➤ YN0013: │ 519 packages were already cached
2023-04-21T17:33:33.6577304Z ➤ YN0000: └ Completed in 0s 340ms
2023-04-21T17:33:33.6911814Z ➤ YN0000: ┌ Link step
2023-04-21T17:33:34.1088182Z ➤ YN0000: └ Completed in 0s 417ms
2023-04-21T17:33:34.1152453Z ➤ YN0000: Done with warnings in 0s 987ms
2023-04-21T17:33:34.1273218Z ##[endgroup]
2023-04-21T17:33:36.8152179Z ##[debug][Telemetry] tracking a 'ScanStart' event
2023-04-21T17:33:36.8159962Z ##[debug][Telemetry] tracking a 'Inputs' event
2023-04-21T17:33:36.8161410Z ##[debug][Telemetry] tracking a 'AuthUsed' event
2023-04-21T17:33:36.8162132Z ##[group]Scanning URL https://portal.azure.com
2023-04-21T17:33:36.8162849Z ##[debug]Starting accessibility scanning of URL https://portal.azure.com
2023-04-21T17:33:36.8163692Z ##[debug]Chrome app executable: /usr/bin/google-chrome
2023-04-21T17:33:36.8165059Z ##[debug] Created a data storage folder at /mnt/vss/_work/_temp/accessibility-insights-action-6jhpsw. You can override the path by setting the APIFY_LOCAL_STORAGE_DIR environment variable
2023-04-21T17:33:36.8166241Z ##[debug][33m PuppeteerCrawler:AutoscaledPool:Snapshotter:[39m Setting max memory of this run to 1988 MB. Use the APIFY_MEMORY_MBYTES environment variable to override it.
2023-04-21T17:33:36.8167830Z ##[debug][33m PuppeteerCrawler:AutoscaledPool:[39m state[90m {"currentConcurrency":0,"desiredConcurrency":1,"systemStatus":{"isSystemIdle":true,"memInfo":{"isOverloaded":false,"limitRatio":0.2,"actualRatio":null},"eventLoopInfo":{"isOverloaded":false,"limitRatio":0.6,"actualRatio":null},"cpuInfo":{"isOverloaded":false,"limitRatio":0.4,"actualRatio":null},"clientInfo":{"isOverloaded":false,"limitRatio":0.3,"actualRatio":null}}}[39m
2023-04-21T17:33:55.4177366Z ##[debug]KMSI page handled.
2023-04-21T17:33:55.4182444Z ##[debug]Authentication succeeded.
2023-04-21T17:34:04.8722376Z Processing page https://ms.portal.azure.com/#home
2023-04-21T17:34:04.8922955Z [31mERROR[39m[33m PuppeteerCrawler:[39m handleRequestFunction failed, reclaiming failed request back to the list or queue[90m {"url":"https://portal.azure.com","retryCount":1,"id":"uhzXA4X8fGqJ7gN"}[39m
2023-04-21T17:34:04.8946049Z   Error: Evaluation failed: TypeError: pQuerySelectorAll is not a function
2023-04-21T17:34:04.8948224Z       at pptr://__puppeteer_evaluation_script__:2:12
2023-04-21T17:34:04.8949879Z       at ExecutionContext._ExecutionContext_evaluate (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ExecutionContext.js:252:15)
2023-04-21T17:34:04.8951083Z       at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-04-21T17:34:04.8952457Z       at async CDPJSHandle.evaluateHandle (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/JSHandle.js:65:16)
2023-04-21T17:34:04.9003632Z       at async Function.queryAll (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/QueryHandler.js:73:24)
2023-04-21T17:34:04.9004981Z       at async Function.collect (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/util/AsyncIterableUtil.js:20:26)
2023-04-21T17:34:04.9007651Z       at async CDPElementHandle.$$eval (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ElementHandle.js:115:25)
2023-04-21T17:34:04.9009207Z       at async ClassicPageProcessor.enqueueLinks [as enqueueLinksExt] (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/apify/build/enqueue_links/enqueue_links.js:112:25)
2023-04-21T17:34:04.9010588Z       at async ClassicPageProcessor.enqueueLinks (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4425:26)
2023-04-21T17:34:04.9011841Z       at async ClassicPageProcessor.processPage (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4263:13)
2023-04-21T17:34:04.9013148Z       at async PuppeteerCrawler.PageProcessorBase.pageHandler [as handlePageFunction] (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4338:17)
2023-04-21T17:34:16.2056825Z Processing page https://ms.portal.azure.com/#home
2023-04-21T17:34:16.3233201Z [31mERROR[39m[33m PuppeteerCrawler:[39m handleRequestFunction failed, reclaiming failed request back to the list or queue[90m {"url":"https://portal.azure.com","retryCount":2,"id":"uhzXA4X8fGqJ7gN"}[39m
2023-04-21T17:34:16.3235588Z   Error: Evaluation failed: TypeError: pQuerySelectorAll is not a function
2023-04-21T17:34:16.3346298Z       at pptr://__puppeteer_evaluation_script__:2:12
2023-04-21T17:34:16.3348181Z       at ExecutionContext._ExecutionContext_evaluate (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ExecutionContext.js:252:15)
2023-04-21T17:34:16.3349007Z       at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-04-21T17:34:16.3350032Z       at async CDPJSHandle.evaluateHandle (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/JSHandle.js:65:16)
2023-04-21T17:34:16.3351387Z       at async Function.queryAll (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/QueryHandler.js:73:24)
2023-04-21T17:34:16.3352436Z       at async Function.collect (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/util/AsyncIterableUtil.js:20:26)
2023-04-21T17:34:16.3353479Z       at async CDPElementHandle.$$eval (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ElementHandle.js:115:25)
2023-04-21T17:34:16.3354540Z       at async ClassicPageProcessor.enqueueLinks [as enqueueLinksExt] (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/apify/build/enqueue_links/enqueue_links.js:112:25)
2023-04-21T17:34:16.3355546Z       at async ClassicPageProcessor.enqueueLinks (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4425:26)
2023-04-21T17:34:16.3356541Z       at async ClassicPageProcessor.processPage (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4263:13)
2023-04-21T17:34:16.3357831Z       at async PuppeteerCrawler.PageProcessorBase.pageHandler [as handlePageFunction] (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4338:17)
2023-04-21T17:34:29.5812393Z Processing page https://ms.portal.azure.com/#home
2023-04-21T17:34:29.5928967Z [31mERROR[39m[33m PuppeteerCrawler:[39m handleRequestFunction failed, reclaiming failed request back to the list or queue[90m {"url":"https://portal.azure.com","retryCount":3,"id":"uhzXA4X8fGqJ7gN"}[39m
2023-04-21T17:34:29.5930230Z   Error: Evaluation failed: TypeError: pQuerySelectorAll is not a function
2023-04-21T17:34:29.5930778Z       at pptr://__puppeteer_evaluation_script__:2:12
2023-04-21T17:34:29.5932030Z       at ExecutionContext._ExecutionContext_evaluate (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ExecutionContext.js:252:15)
2023-04-21T17:34:29.5932868Z       at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-04-21T17:34:29.5934006Z       at async CDPJSHandle.evaluateHandle (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/JSHandle.js:65:16)
2023-04-21T17:34:29.5935157Z       at async Function.queryAll (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/QueryHandler.js:73:24)
2023-04-21T17:34:29.5936565Z       at async Function.collect (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/util/AsyncIterableUtil.js:20:26)
2023-04-21T17:34:29.5937638Z       at async CDPElementHandle.$$eval (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ElementHandle.js:115:25)
2023-04-21T17:34:29.5938940Z       at async ClassicPageProcessor.enqueueLinks [as enqueueLinksExt] (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/apify/build/enqueue_links/enqueue_links.js:112:25)
2023-04-21T17:34:29.5939934Z       at async ClassicPageProcessor.enqueueLinks (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4425:26)
2023-04-21T17:34:29.5940865Z       at async ClassicPageProcessor.processPage (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4263:13)
2023-04-21T17:34:29.5941978Z       at async PuppeteerCrawler.PageProcessorBase.pageHandler [as handlePageFunction] (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4338:17)
2023-04-21T17:34:35.1823898Z ##[debug][33m PuppeteerCrawler:[39m All the requests from request list and/or request queue have been processed, the crawler will shut down.
2023-04-21T17:34:35.1863068Z ##[debug][33m PuppeteerCrawler:[39m Final request statistics:[90m {"requestsFinished":0,"requestsFailed":1,"retryHistogram":[null,null,null,1],"requestAvgFailedDurationMillis":2064,"requestAvgFinishedDurationMillis":null,"requestsFinishedPerMinute":0,"requestsFailedPerMinute":1,"requestTotalDurationMillis":2064,"requestsTotal":1,"crawlerRuntimeMillis":59200}[39m
2023-04-21T17:34:35.1926478Z ##[error]Scan failed with 1 error(s)
2023-04-21T17:34:35.1936165Z ##[error][Exception]ErrorWithCause: An error occurred while scanning website page: https://portal.azure.com
2023-04-21T17:34:35.1937827Z     at Logger.trackExceptionAny (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:187845:29)
2023-04-21T17:34:35.1941262Z     at Scanner.logAndTrackScanningException (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:189115:21)
2023-04-21T17:34:35.1942704Z     at /mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:189046:30
2023-04-21T17:34:35.1943644Z     at Array.forEach (<anonymous>)
2023-04-21T17:34:35.2008498Z     at Scanner.<anonymous> (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:189045:47)
2023-04-21T17:34:35.2009312Z     at Generator.next (<anonymous>)
2023-04-21T17:34:35.2010153Z     at fulfilled (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:188948:58)
2023-04-21T17:34:35.2010802Z     at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-04-21T17:34:35.2011380Z caused by: Error: '{\n' +
2023-04-21T17:34:35.2011862Z   "  name: 'Error',\n" +
2023-04-21T17:34:35.2012520Z   "  message: 'Execution context was destroyed, most likely because of a navigation.',\n" +
2023-04-21T17:34:35.2013327Z   "  stack: 'Error: Execution context was destroyed, most likely because of a navigation.\\n' +\n" +
2023-04-21T17:34:35.2014461Z   "    '    at rewriteError (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ExecutionContext.js:309:15)\\n' +\n" +
2023-04-21T17:34:35.2015407Z   "    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)\\n' +\n" +
2023-04-21T17:34:35.2016507Z   "    '    at async ExecutionContext._ExecutionContext_evaluate (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ExecutionContext.js:216:60)\\n' +\n" +
2023-04-21T17:34:35.2018222Z   "    '    at async ExecutionContext.convertArgument (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ExecutionContext.js:259:19)\\n' +\n" +
2023-04-21T17:34:35.2019326Z   "    '    at async Promise.all (index 0)\\n' +\n" +
2023-04-21T17:34:35.2020444Z   "    '    at async ExecutionContext._ExecutionContext_evaluate (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ExecutionContext.js:237:24)\\n' +\n" +
2023-04-21T17:34:35.2021755Z   "    '    at async Frame.addStyleTag (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/puppeteer-core/lib/cjs/puppeteer/common/Frame.js:626:75)\\n' +\n" +
2023-04-21T17:34:35.2023016Z   "    '    at async PageNavigationHooks.disableAnimation (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:7210:9)\\n' +\n" +
2023-04-21T17:34:35.2024344Z   "    '    at async PageNavigationHooks.postNavigation (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:7185:9)\\n' +\n" +
2023-04-21T17:34:35.2025633Z   "    '    at async PageProcessorBase.postNavigation (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/node_modules/accessibility-insights-scan/dist/index.js:4360:17)'\n" +
2023-04-21T17:34:35.2026375Z   '}'
2023-04-21T17:34:35.2027164Z     at Logger.trackExceptionAny (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:187844:94)
2023-04-21T17:34:35.2028210Z     at Scanner.logAndTrackScanningException (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:189115:21)
2023-04-21T17:34:35.2029210Z     at /mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:189046:30
2023-04-21T17:34:35.2029823Z     at Array.forEach (<anonymous>)
2023-04-21T17:34:35.2030654Z     at Scanner.<anonymous> (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:189045:47)
2023-04-21T17:34:35.2031248Z     at Generator.next (<anonymous>)
2023-04-21T17:34:35.2032039Z     at fulfilled (/mnt/vss/_work/_tasks/accessibility-insights_aebf5565-d905-5999-8faf-a44d4db6c0fa/3.0.9/index.js:188948:58)
2023-04-21T17:34:35.2032698Z     at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-04-21T17:34:35.2033354Z ##[debug][Telemetry] tracking a 'ErrorFound' event
2023-04-21T17:34:35.2033845Z Accessibility scanning of URL https://portal.azure.com completed
2023-04-21T17:34:35.2749511Z ##[error]Scan failed with 1 error(s)
2023-04-21T17:34:35.8350216Z ##[section]Finishing: case 12: url, azure portal auth test
```


## Additional context

This bug has been present in the ADO extension since it upgraded the accessibility-insights-scan package from version 1.6.0 to version 1.7.0. This issue is being tracked in the extension as well: https://github.com/microsoft/accessibility-insights-action/issues/1713

--- a/.vscode/workspace.code-workspace
+++ b/.vscode/workspace.code-workspace
@@ -142,6 +142,7 @@
         "files.eol": "\n",
         "files.insertFinalNewline": true,
         "prettier.requireConfig": true,
+        "csharp.suppressDotnetRestoreNotification": true,
         "eslint.enable": true,
         "eslint.packageManager": "yarn",
         "eslint.alwaysShowStatus": true,
@@ -190,8 +191,7 @@
             "typemoq",
             "unscannable",
             "upsert"
-        ],
-        "csharp.suppressDotnetRestoreNotification": true
+        ]
     },
     "extensions": {
         "recommendations": [
@@ -203,7 +203,8 @@
             "streetsidesoftware.code-spell-checker",
             "ms-azuretools.vscode-azurefunctions",
             "christian-kohler.npm-intellisense",
-            "ms-vscode.powershell"
+            "ms-vscode.powershell",
+            "ms-azuretools.vscode-docker"
         ]
     },
 }

--- a/packages/cli/README.md
+++ b/packages/cli/README.md
@@ -128,7 +128,7 @@ default: false
 
 ```sh
 type: array
-describe: List of CSS selectors to match against, separated by space. Default selector is 'button'.
+describe: List of CSS selectors to match against. Default selector is 'button'.
 default: ['button']
 ```
 
@@ -199,14 +199,14 @@ describe: File path that contains list of URLs (each separated by a new line) to
 
 ```sh
 type: array
-describe: List of URLs to crawl in addition to URLs discovered from crawling the provided URL, separated by space.
+describe: List of URLs to crawl in addition to URLs discovered from crawling the provided URL.
 ```
 
 -   discoveryPatterns: --discoveryPatterns
 
 ```sh
 type: array
-describe: List of RegEx patterns to crawl in addition to the provided URL, separated by space.
+describe: List of RegEx patterns to crawl in addition to the provided URL.
 ```
 
 -   baselineFile: --baselineFile
@@ -288,5 +288,5 @@ The URL folder is a resource location equal to base URL up-to the last forward s
 
 ```sh
 type: array
-describe: The Chrome command line option to pass on browser start. Can be used to disable CORS to scan protected page: --browserOptions disable-web-security
+describe: List of Chrome command line options to pass on browser start. Can be used to disable CORS to scan protected page: --browserOptions disable-web-security
 ```

--- a/packages/cli/README.md
+++ b/packages/cli/README.md
@@ -283,3 +283,10 @@ The URL folder is a resource location equal to base URL up-to the last forward s
     -   If base URL is specified as https://www.example.com/bar/foo , URLs that are in https://www.example.com/bar/ folder will be considered for crawling and scanning.
     -   If base URL is specified as https://www.example.com/bar/foo/ , only URLs that are in https://www.example.com/bar/foo/ folder will be considered for crawling and scanning.
 ```
+
+-   browserOptions: --browserOptions
+
+```sh
+type: array
+describe: The Chrome command line option to pass on browser start. Can be used to disable CORS to scan protected page: --browserOptions disable-web-security
+```

--- a/packages/cli/package.json
+++ b/packages/cli/package.json
@@ -1,6 +1,6 @@
 {
     "name": "accessibility-insights-scan",
-    "version": "2.4.0",
+    "version": "2.4.1",
     "description": "This project welcomes contributions and suggestions.  Most contributions require you to agree to a Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us the rights to use your contribution. For details, visit https://cla.microsoft.com.",
     "scripts": {
         "build": "webpack --config ./webpack.config.js \"$@\"",
@@ -47,9 +47,7 @@
         "mkdirp": "^2.1.3",
         "mockdate": "^3.0.5",
         "npm-run-all": "^4.1.5",
-        "playwright": "^1.30.0",
         "rimraf": "^4.1.2",
-        "shebang-loader": "^0.0.1",
         "ts-jest": "^29.1.0",
         "typemoq": "^2.1.0",
         "typescript": "^4.9.5",

--- a/packages/cli/src/cli.ts
+++ b/packages/cli/src/cli.ts
@@ -46,7 +46,7 @@ function getScanArguments(): ScanArguments {
             },
             selectors: {
                 type: 'array',
-                describe: `List of CSS selectors to match against, separated by space. Default selector is 'button'.`,
+                describe: `List of CSS selectors to match against. Default selector is 'button'.`,
                 default: [],
             },
             output: {
@@ -95,12 +95,12 @@ function getScanArguments(): ScanArguments {
             },
             inputUrls: {
                 type: 'array',
-                describe: `List of URLs to crawl in addition to URLs discovered from crawling the provided URL, separated by space.`,
+                describe: `List of URLs to crawl in addition to URLs discovered from crawling the provided URL.`,
                 alias: 'inputurls',
             },
             discoveryPatterns: {
                 type: 'array',
-                describe: `List of RegEx patterns to crawl in addition to the provided URL, separated by space.`,
+                describe: `List of RegEx patterns to crawl in addition to the provided URL.`,
                 alias: 'discoverypatterns',
             },
             baselineFile: {
@@ -160,7 +160,7 @@ function getScanArguments(): ScanArguments {
             },
             browserOptions: {
                 type: 'array',
-                describe: `The Chrome command line option to pass on browser start. Can be used to disable CORS to scan protected page: --browserOptions disable-web-security`,
+                describe: `List of Chrome command line options to pass on browser start. Can be used to disable CORS to scan protected page: --browserOptions disable-web-security`,
                 alias: 'browseroptions',
             },
         })

--- a/packages/cli/src/cli.ts
+++ b/packages/cli/src/cli.ts
@@ -158,6 +158,11 @@ function getScanArguments(): ScanArguments {
                     'Adhere to the pattern when URLs with a trailing slash indicates a directory, and those without a trailing slash denotes a file.',
                 default: true,
             },
+            browserOptions: {
+                type: 'array',
+                describe: `The Chrome command line option to pass on browser start. Can be used to disable CORS to scan protected page: --browserOptions disable-web-security`,
+                alias: 'browseroptions',
+            },
         })
         .check((args) => {
             validateScanArguments(args as ScanArguments);

--- a/packages/cli/src/crawler/crawler-parameters-builder.ts
+++ b/packages/cli/src/crawler/crawler-parameters-builder.ts
@@ -62,6 +62,7 @@ export class CrawlerParametersBuilder {
             authType: scanArguments.authType,
             httpHeaders: headers,
             adhereFilesystemPattern: scanArguments.adhereFilesystemPattern,
+            browserOptions: scanArguments.browserOptions,
         };
     }
 

--- a/packages/cli/src/scan-arguments.ts
+++ b/packages/cli/src/scan-arguments.ts
@@ -29,4 +29,5 @@ export interface ScanArguments {
     userAgent?: string;
     httpHeaders?: string;
     adhereFilesystemPattern?: boolean;
+    browserOptions?: string[];
 }

--- a/packages/crawler/src/crawler/site-crawler-engine.ts
+++ b/packages/crawler/src/crawler/site-crawler-engine.ts
@@ -34,6 +34,7 @@ export class SiteCrawlerEngine implements CrawlerEngine {
         this.crawlerConfiguration.setMemoryMBytes(crawlerRunOptions.memoryMBytes);
         this.crawlerConfiguration.setSilentMode(crawlerRunOptions.silentMode);
 
+        const puppeteerOptions = crawlerRunOptions.browserOptions ? crawlerRunOptions.browserOptions.map((o) => `--${o}`) : [];
         const puppeteerDefaultOptions = [
             '--disable-dev-shm-usage',
             '--no-sandbox',
@@ -52,7 +53,7 @@ export class SiteCrawlerEngine implements CrawlerEngine {
             maxRequestsPerCrawl: this.crawlerConfiguration.maxRequestsPerCrawl(),
             launchContext: {
                 launchOptions: {
-                    args: puppeteerDefaultOptions,
+                    args: [...puppeteerDefaultOptions, ...puppeteerOptions],
                     defaultViewport: {
                         ...windowSize,
                         deviceScaleFactor: 1,

--- a/packages/crawler/src/logger/apify-console-logger-client.ts
+++ b/packages/crawler/src/logger/apify-console-logger-client.ts
@@ -12,7 +12,7 @@ import { LogLevel, LoggerClient, LoggerProperties } from './logger';
 export class ApifyConsoleLoggerClient implements LoggerClient {
     private baseProperties: LoggerProperties;
 
-    constructor(private readonly logger: Log = new Log({ prefix: 'Crawler' })) {}
+    constructor(private readonly logger: Log = new Log({ prefix: 'Scanner' })) {}
 
     public log(message: string, logLevel: LogLevel, properties?: LoggerProperties): void {
         const logMessage = `[${moment.utc().toISOString()}] ${message}`;

--- a/packages/crawler/src/main.ts
+++ b/packages/crawler/src/main.ts
@@ -26,7 +26,20 @@ interface ScanArguments {
  */
 (async () => {
     dotenv.config();
-    const scanArguments = yargs.argv as unknown as ScanArguments;
+    const scanArguments = yargs.option({
+        selectors: {
+            type: 'array',
+        },
+        inputUrls: {
+            type: 'array',
+        },
+        discoveryPatterns: {
+            type: 'array',
+        },
+        browserOptions: {
+            type: 'array',
+        },
+    }).argv as unknown as ScanArguments;
 
     const container = new inversify.Container({ autoBindInjectable: true });
     setupCrawlerContainer(container);

--- a/packages/crawler/src/page-handler/page-response-processor.spec.ts
+++ b/packages/crawler/src/page-handler/page-response-processor.spec.ts
@@ -35,7 +35,7 @@ describe(PageResponseProcessor, () => {
         responseMock
             .setup((o) => o.status())
             .returns(() => 404)
-            .verifiable();
+            .verifiable(Times.atLeast(2));
 
         const expectedError = {
             errorType: 'HttpErrorCode',
@@ -48,6 +48,27 @@ describe(PageResponseProcessor, () => {
         const actualError = pageResponseProcessor.getResponseError(responseMock.object, { stack: 'stack' } as Error);
 
         expect(actualError).toEqual(expectedError);
+    });
+
+    it('get response for HTTP 304 response code', () => {
+        responseMock
+            .setup((o) => o.ok())
+            .returns(() => false)
+            .verifiable();
+        responseMock
+            .setup((o) => o.status())
+            .returns(() => 304)
+            .verifiable(Times.atLeast(2));
+        responseMock
+            .setup((o) => o.headers())
+            .returns(() => {
+                return {}; // The HTTP 304 browser response has no content-type header
+            })
+            .verifiable();
+
+        const actualError = pageResponseProcessor.getResponseError(responseMock.object);
+
+        expect(actualError).toBeUndefined();
     });
 
     it('get response error for invalid content type', () => {

--- a/packages/crawler/src/page-handler/page-response-processor.ts
+++ b/packages/crawler/src/page-handler/page-response-processor.ts
@@ -10,7 +10,7 @@ export class PageResponseProcessor {
     constructor(private readonly navigationErrorPatterns: Partial<Record<BrowserErrorTypes, string[]>> = pageNavigationErrorPatterns) {}
 
     public getResponseError(response: Puppeteer.HTTPResponse, error: Error = new Error()): BrowserError {
-        if (!response.ok()) {
+        if (response.ok() !== true && response.status() !== 304) {
             return {
                 errorType: 'HttpErrorCode',
                 statusCode: response.status(),
@@ -20,7 +20,8 @@ export class PageResponseProcessor {
             };
         }
 
-        if (!this.isHtmlContentType(response)) {
+        // The HTTP 304 browser response has no content-type header
+        if (this.isHtmlContentType(response) !== true && response.status() !== 304) {
             const contentType = this.getContentType(response.headers());
 
             return {

--- a/packages/crawler/src/page-processors/page-processor-base.ts
+++ b/packages/crawler/src/page-processors/page-processor-base.ts
@@ -143,10 +143,24 @@ export abstract class PageProcessorBase implements PageProcessor {
         // converting relative href link to absolute link.
         context.request.loadedUrl = context.page.url();
 
-        const enqueued = await context.enqueueLinks({
-            // eslint-disable-next-line security/detect-non-literal-regexp
-            regexps: this.discoveryPatterns?.length > 0 ? this.discoveryPatterns.map((p) => new RegExp(p)) : undefined,
-        });
+        let enqueued;
+        try {
+            enqueued = await context.enqueueLinks({
+                // eslint-disable-next-line security/detect-non-literal-regexp
+                regexps: this.discoveryPatterns?.length > 0 ? this.discoveryPatterns.map((p) => new RegExp(p)) : undefined,
+            });
+        } catch (error) {
+            if ((error as Error).message?.includes('pQuerySelectorAll is not a function')) {
+                throw new Error(
+                    `Puppeteer has failed to inject an automation script due to page security settings. Try to use disable-web-security browser option to scan page. ${System.serializeError(
+                        error,
+                    )}`,
+                );
+            }
+
+            throw error;
+        }
+
         this.logger.logInfo(`Enqueued ${enqueued.processedRequests.length} new links.`, {
             url: context.page.url(),
         });

--- a/packages/crawler/src/types/crawler-run-options.ts
+++ b/packages/crawler/src/types/crawler-run-options.ts
@@ -26,4 +26,5 @@ export interface CrawlerRunOptions {
     authType?: string;
     httpHeaders?: Record<string, string>;
     adhereFilesystemPattern?: boolean;
+    browserOptions?: string[];
 }

